src/hal/sample_channel/sample_channel.hal

```diff
@@ -9,13 +9,17 @@ loadrt siggen
 # tell max cycles the ring should be able to contain by: cycles=100
 newinst sample_channel_pb sampler --- samples=bfusUSf cycles=100
 
+newinst sample_channel_pb lcec.0.1 --- samples=bff pinnames=food,bar,bazooka cycles=100
```

Comment:
The third `-` is probably being passed as an arg which is why it mostly works.
Should only need 2 `--` for hal_command to separate off the strings after them and pass as args to the component.

Reply:
hmm, I've taken the debug output from both instaces, I had seen this but not made the link that this was not meant to be. I've always used 3 dashes, probably slipped in a typo after one of the first examples I've seen when it got introduced.
```
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt halg_xinitfv:90 HAL: initializing component 'sample_channel_pb' type=1 arg1=0 arg2=0/0x0
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: sampler: argument 2 = ---
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: sampler: argument 3 = samples=bfusUSf
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: sampler: argument 4 = cycles=100
...
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: lcec.0.1: argument 2 = ---
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: lcec.0.1: argument 3 = samples=bff
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: lcec.0.1: argument 4 = pinnames=food,bar,bazooka
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt pinnames = food,bar,bazooka
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt sample_channel_pb: lcec.0.1: argument 5 = cycles=100
Jun  9 13:44:56 TP51 msgd:0: hal_lib:1807:rt cycles = 100
```
and indeed, changing the `---` to `--` shows:
```
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt halg_xinitfv:90 HAL: initializing component 'sample_channel_pb' type=1 arg1=0 arg2=0/0x0
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt sample_channel_pb: sampler: argument 2 = samples=bfusUSf
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt sample_channel_pb: sampler: argument 3 = cycles=100
...
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt sample_channel_pb: lcec.0.1: argument 2 = samples=bff
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt sample_channel_pb: lcec.0.1: argument 3 = pinnames=food,bar,bazooka
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt pinnames = food,bar,bazooka
Jun  9 13:55:11 TP51 msgd:0: hal_lib:3352:rt sample_channel_pb: lcec.0.1: argument 4 = cycles=100
```
i'll fix the hal file